egs/wsj/s5/steps/cleanup/do_cleanup_segmentation.sh

```diff
@@ -0,0 +1,147 @@
+#!/bin/bash
+
+# Copyright 2016  Vimal Manohar
+# Apache 2.0
+
+# This script demonstrates how to re-segment training data selecting only the 
+# "good" audio that matches the transcripts.
+# The basic idea is to decode with an existing in-domain acoustic model, and a
+# biased language model built from the reference, and then work out the
+# segmentation from a ctm like file.
+
+set -e 
+set -o pipefail 
+
+stage=0
+
+pad_length=5                  # Number of frames for padding the created 
+                              # subsegments
+max_silence_length=50         # Maxium number of silence frames above which they are removed and the segment is split
+max_incorrect_words=2         # Maximum number of incorrect words allowed in the segments
+min_correct_frames=0          # Minimum number of frames required in a subsegment to be kept
+max_segment_wer=20000         # Maximum WER% of an utterance, above which it is simply removed
+min_wer_for_splitting=-1      # Minimum WER% for a segment to be considered for splitting
+
+group_utt_words=              # If provided, build LMs on group of utterances 
+                              # that have at least these many words.  
+                              # Otherwise, build per-utterance LM
+
+# ngram options
+ngram_order=2                 
+interpolation_weights=0:0.1:0.9       # LM combination weights for 0-gram:1-gram:2-gram
+top_words_interpolation_weight=0.1    # Interpolation weight for top-words unigram
+top_n_words=100                       # Number of top-words to use for building a unigram LM
+
+lmwt=10         
+
+cmd=run.pl
+nj=4
+
+. ./path.sh
+. utils/parse_options.sh 
+
+
+if [ $# -ne 6 ]; then
+  echo "Usage: $0 <data> <lang> <srcdir> <dir> <segmentation-dir> <out-data>"
+  echo " e.g.: $0 data/sdm1/train_sp data/lang exp/sdm1/tri4a_ali_train_sp exp/sdm1/tri4a_sdm1_train_sp_bad_utts exp/sdm1/tri4a_sdm1/train_sp_bad_utts/segmentation_cleaned data/sdm1/train_sp_cleaned"
+  exit 1
+fi
+
+data=$1
+lang=$2
+srcdir=$3
+dir=$4
+segmentation_dir=$5
+out_data=$6
+
+dataid=`basename $data`
+
+mkdir -p $dir
+cp $srcdir/final.mdl $dir
+cp $srcdir/tree $dir
+cp $srcdir/cmvn_opts $dir
+cp $srcdir/{splice_opts,final.mat,final.alimdl} $dir || true
+chmod +w $dir/{final.mdl,tree,cmvn_opts}
```

Comment:
Some chain scripts used to make these non-writable. But it might not be relevant here anymore.